jsonpatcher/jsonpatcher.go

```diff
@@ -11,9 +11,9 @@ import (
 	jsonpatch "github.com/evanphx/json-patch"
```

Comment:
Diff in this file is quite insane diff in this file... pretty hard to understand.

A bit of a summary here:
- Instead of `WrapObject` the json-patches (which were single changes), now what's `WrapObject` is a new type:
```
type recordEvents struct {
	Patches []patchEvent
}
```
which basically is a set of `patchEvents`.

In the way of making this change, I avoided some json-serializations and some other things that makes things clearer (might be better to see the direct file and not the diff).

Most of this code is the heart of the main PR intention.

Reply:
Awesome